app/src/services/objectPermission.js

```diff
@@ -61,6 +61,24 @@ const service = {
     }
   },
 
+  /**
+   * @function getObjectIdsWithBucket
+   * Searches for specific (object) bucket permissions
+   * @param {string|string[]} [params.userId] Optional string or array of uuids representing the user
+   * @param {string|string[]} [params.bucketId] Optional string or array of bucket id(s)
+   * @returns {Promise<object>} The result of running the find operation
+  */
+  getObjectIdsWithBucket: async (userId, bucketId) => {
+    return ObjectPermission.query()
```

Comment:
here we want to get 'all objects that are in buckets that user has any bucket permission for'. 
i believe we can do:
```js
    return BucketPermission.query()
   .rightJoin('object', 'bucket_permission.bucketId', '=', 'object.bucketId')
  .select('object.objectId')
  .distinct('userId')
  .modify('filterUserId', userId)
  .modify('filterBucketId', bucketId)
  .whereNotNull('object.bucketId')
  .then(response => response.map(entry => entry.objectId));